app/uk/gov/hmrc/customs/emailfrontend/services/EmailCacheService.scala

```diff
@@ -60,19 +61,33 @@ class EmailCacheService @Inject()(caching: Save4LaterCachingConfig, applicationC
   }
 
   def saveTimeStamp(internalId: InternalId, verifiedEmailTimestamp: DateTime)
-               (implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[CacheMap] = {
+                   (implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[CacheMap] = {
+
     import uk.gov.hmrc.customs.emailfrontend.DateTimeUtil._
     Logger.info("saving verified email time stamp to save 4 later")
     cache[DateTime](internalId.id, timestampKey, verifiedEmailTimestamp)
   }
 
+  private def fetchTimeStamp(internalId: InternalId)(implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[Option[DateTime]] = {
+    import uk.gov.hmrc.customs.emailfrontend.DateTimeUtil._
+    Logger.info("calling save 4 later to retrieve timestamp")
+    fetchAndGetEntry[DateTime](internalId.id, timestampKey)
+  }
+
+  def emailVerificationStatus(internalId: InternalId)(implicit hc: HeaderCarrier, executionContext: ExecutionContext): Future[EmailAmendmentStatus] = {
```

Comment:
shouldn't this be renamed and moved to controller/service specific to amendment. As this service class is all about save4later cache functions. Also, is it worth renaming the class name as well?

Reply:
Also, `fetchTimeStamp` could be tested independently and not making that as private function. Since, it's like an utility function providing access to save4later @shaileshp0110

Reply:
yes but for now let's keep it private if it's needed in future we can expose this method
emailAmendmentStatus @kapil-sanghvi. Unless you want this change now only I am bit in dual status on this

Reply:
I suggest to leave this for now. Maybe in the near future opportunity will come to deal with it with some better idea.

Reply:
I would have preferred it with this one, since you have got unit tests covering this functionality via another function. So, you left it for someone else to refactor all those tests when they really need to only use `fetchTimeStamp` functionality..disagree with @Jodlowy

Reply:
And it would make more sense to have `emailAmendmentStatus` function in the controller rather than having in `save4later` functionality provider class